### PR TITLE
update target sdk version to 31

### PR DESCRIPTION
**Pull request by pt2121**
As mentioned in https://github.com/grote/Transportr/issues/852#issuecomment-1551918703, Google Play new requires the target SDK to be at least 31. Also see https://support.google.com/googleplay/android-developer/answer/11926878.

I went through [the doc here](https://developer.android.com/google/play/requirements/target-sdk#pre12) & [here](https://developer.android.com/about/versions/12/behavior-changes-12), and made sure that we updated everything that we should - fortunately it's not that much. From what I see, we don't use any foreground service, notif, BroadcastReceiver so it seems like a smooth update :crossed_fingers:

### CODE REVIEW

**Comment by pt2121** [comment]
@cla-bot I accept the CLA.

**Comment by Altonss** [comment]
@cla-bot check

**Comment by Altonss** [comment]
@cla-bot[bot] check

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -96,6 +97,10 @@
 			android:name=".about.ContributorsActivity"
 			android:label="@string/drawer_contributors"/>
 
+		<service
+			android:name="com.mapzen.android.lost.internal.GeofencingIntentService"
+			android:exported="true" />
```
Comment (ialokim) [inline app/src/main/AndroidManifest.xml:102]: Could you explain why this is needed now?

Comment (pt2121) [reply app/src/main/AndroidManifest.xml:102]: Good question. Basically, for Android 12+, any Android components with intent filters require to explicitly declare the `exported`. More info is [here](https://developer.android.com/about/versions/12/behavior-changes-12#exported).
My understanding is that we use this service internally and I found a solution [here](https://github.com/lostzen/lost/issues/272)

**Comment by Altonss** [comment]
Why not target level 33 directly? Is this significantly more work? I looked here https://developer.android.com/google/play/requirements/target-sdk and found that starting of August, 33 will be required by PlayStore

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (grote) [reply app/src/main/AndroidManifest.xml:102]: Does it work without exporting the service? i.e. setting exported to false

**Review by grote — APPROVED** [review]

**Pull request merged**